backend/coins/btc/account_test.go

```diff
@@ -95,4 +109,39 @@ func TestAccount(t *testing.T) {
 	require.Equal(t, accounts.OrderedTransactions{}, transactions)
 
 	require.Equal(t, []*btc.SpendableOutput{}, account.SpendableOutputs())
+
+	// check the number of available addresses for native and wrapped segwit.
+	require.Equal(t, len(account.GetUnusedReceiveAddresses()[0].Addresses), 20)
+	require.Equal(t, *account.GetUnusedReceiveAddresses()[0].ScriptType, signing.ScriptTypeP2WPKHP2SH)
+	require.Equal(t, len(account.GetUnusedReceiveAddresses()[1].Addresses), 20)
+	require.Equal(t, *account.GetUnusedReceiveAddresses()[1].ScriptType, signing.ScriptTypeP2WPKH)
+
+	// Create a new insured account.
+	account2 := btc.NewAccount(
+		&accounts.AccountConfig{
+			Config: &config.Account{
+				Code:                  "accountcode2",
+				Name:                  "accountname2",
+				SigningConfigurations: signingConfigurations,
+				InsuranceStatus:       "active",
+			},
+			DBFolder:        dbFolder,
+			OnEvent:         func(accountsTypes.Event) {},
+			RateUpdater:     nil,
+			GetNotifier:     func(signing.Configurations) accounts.Notifier { return nil },
+			GetSaveFilename: func(suggestedFilename string) string { return suggestedFilename },
+		},
+		coin, nil,
+		logging.Get().WithGroup("account_test"),
+	)
+	require.NoError(t, account2.Initialize())
+	require.True(t, account2.Synced())
+
+	// verify that the wrapped segwit addresses are not available.
+	require.Equal(t, len(account2.GetUnusedReceiveAddresses()[0].Addresses), 0)
```

Comment:
why even provide it at all, why does `GetUnusedReceiveAddresses()` not return only one slice element for native segwit?

Reply:
Yeah, you're right, it was a bit lazy as it was before. Fixed.